libbeat/common/transport/tlscommon/tls_config_test.go

```diff
@@ -49,15 +47,15 @@ func TestMakeVerifyServerConnection(t *testing.T) {
 		peerCerts        []*x509.Certificate
 		serverName       string
 		expectedCallback bool
-		expectedError    error
+		expectedError    bool
```

Comment:
Why did you change this?

Reply:
The validation using this field as an error wasn't quite correct, it used be like this:
```go
assert.Error(t, test.expectedError, err)
```

However `assert.Error`'s signature is: `(t TestingT, err error, msgAndArgs ...interface{}) bool`, hence the test was validating that the test case had an error set and using the returned error as a message.

I did try for a wee bit to get one of the other error functions provided by testify to assert the error type, but they didn't quite work, nor I invested much time on it. In the end it was simpler and quicker to just test for error/no error.

Does it make sense? Do we need to be more specific while checking those errors?

Testify can be quite hard to use...

Reply:
I've improved the tests to also validate the error type and message on bc05440788.